hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/cluster/update/strategy/UpdateStrategy.java

```diff
@@ -50,4 +52,12 @@ public UpdateStrategy(HoodieEngineContext engineContext, HoodieTable table, Set<
    */
   public abstract Pair<I, Set<HoodieFileGroupId>> handleUpdate(I taggedRecordsRDD);
 
+  /**
+   * Check the update records to the file group in clustering with given file group id
+   * NOTE: used in flink engine
+   */
+  public Pair<I, List<HoodieFileGroupId>> handleUpdate(HoodieFileGroupId fileId, I records) {
+    throw new UnsupportedOperationException("Unsupported");
```

Comment:
Can we avoid this, shouldn't the `records` already take the file group id ?

Reply:
Will try to unify the interface.